### PR TITLE
Updated diagnostics per #1099

### LINKED ISSUE #1099: Update Diagnostics
Update the existing Diagnostics article
- Add links to Logging article
- Add Glimpse content and links
- Move exception handling content to error handling (#124)
- Update to latest RC
- Apply latest style guidelines (remove linenos, etc.)

### PR DESCRIPTION
**Pull request by ardalis**
Fixes #1099 
Also adds error-handling stub and cleans up logging slightly to follow no line numbers guidance.

### CODE REVIEW

File: aspnet/fundamentals/diagnostics.rst [hunk 1]
```diff
@@ -3,85 +3,25 @@ Diagnostics
...
-.. note:: Security warning: Enabling detailed error message can leak critical information from your app. You should never enable detailed error messages on a production app.
-
-.. code-block:: html
-
-	<configuration>
-	   <system.web>
-		  <customErrors mode="Off"/>
-	   </system.web>
-	</configuration>
+`Browse or download samples on GitHub <https://github.com/aspnet/Docs/tree/master/aspnet/fundamentals/diagnostics/sample>`_.
```
Comment (Rick-Anderson) [inline aspnet/fundamentals/diagnostics.rst]: Consider moving this to the bottom of the file, under **Additional Resources**
I've been doing that where you don't really need to see the code to read the article.

File: aspnet/fundamentals/diagnostics.rst [hunk 2]
```diff
@@ -91,13 +31,20 @@ The path for this page can be optionally specified in the call to ``UseRuntimeIn
...
 .. code-block:: c#
 
-	app.UseRuntimeInfoPage("/info");
+	if (env.IsDevelopment())
+	{
+	    app.UseRuntimeInfoPage("/info");
+	}
 
-As with ``UseDeveloperExceptionPage()``, it is a good idea to limit public access to diagnostic information about your application. As such, in our sample we are only implementing ``UseRuntimeInfoPage()`` when the EnvironmentName is set to Development.
+It is a good idea to limit public access to diagnostic information about your application. As such, you should only call ``UseRuntimeInfoPage()`` when the Environment is in Development.
```
Comment (Rick-Anderson) [inline aspnet/fundamentals/diagnostics.rst]: Diagnostic information should not be exposed in production. The code above prevents diagnostics outside the development environment.

**Comment by Rick-Anderson** [comment]
looks good :shipit:

File: aspnet/fundamentals/diagnostics.rst (hunk 1 above)
Comment (ardalis) [inline aspnet/fundamentals/diagnostics.rst]: This should probably be consistent within the docs. Is this something we should make a policy, one way or the other? For me as a consumer of docs, I like to be able jump right to the code at the top if I can (or at least, know that I can if I want to as I start reading the article), but I may not be typical.

**Pull request merged**